scripts/release.sh

```diff
@@ -203,35 +203,53 @@ build_container_images_artifacts() {
   while read ad; do
     local arch=${ad%=*}
     local digest=${ad##*=}
-    BUILT_RELEASE_IMAGE_TAGS+=("$(build_release_image "$arch" "@$digest")")
+    local release_image
+    release_image=$(build_release_image "$arch" "@$digest") || return 1
+    BUILT_RELEASE_IMAGE_TAGS+=("$release_image")
   done <<<"$IMAGE_ARCH_DIGESTS"
   echo "${BUILT_RELEASE_IMAGE_TAGS[@]}"
 }
 
 push_container_image_artifacts() {
-  local image_tags="${1}"
+  local image_tags="$1"
   for t in $image_tags; do
     podman push "$t"
   done
 }
 
-push_container_manifest() {
-  local source_tags="$1"
-  local manifest_tag_options=()
-  for t in $source_tags; do
-    manifest_tag_options+=("--amend ${t}")
+podman_create_manifest(){
+  local image_ref_list="${*}"
+  podman manifest create "$IMAGE_REF_BASE:$VERSION" >&2
+  for ref in $image_ref_list; do
+    podman manifest add "$IMAGE_REF_BASE:$VERSION" "$ref"
+  done
+}
+
+docker_create_manifest(){
+  local image_list="${*}"
+  local amend_images_options
+  for image in "${image_list[@]}"; do
+    amend_images_options+="--amend $image"
   done
+  podman manifest create "$IMAGE_REF_BASE:$VERSION" "${image_list[@]}" >&2
+}
```

Comment:
what is the difference between podman and docker images here?

Reply:
Podman and docker diverge slightly on the `manifest create` command.  `docker manifest create` takes a manifest name, and a list of other manifests.  You must add images by prefixing them with the `--amend` option.   Podman only allows you to pass one image to the `create` command inline. More images have to be added via `podman manifest add $IMAGE`. Docker does not have a `manifest add` command.

```
❯ docker manifest create --help

Usage:  docker manifest create MANIFEST_LIST MANIFEST [MANIFEST...]

Create a local manifest list for annotating and pushing to a registry

Options:
  -a, --amend      Amend an existing manifest list
      --insecure   Allow communication with an insecure registry
```

```
[root@fedora ~]# podman manifest create --help
Create manifest list or image index

Description:
  Creates manifest lists or image indexes.

Usage:
  podman manifest create [options] LIST [IMAGE]

Examples:
  podman manifest create mylist:v1.11
  podman manifest create mylist:v1.11 arch-specific-image-to-add
  podman manifest create --all mylist:v1.11 transport:tagged-image-to-add

Options:
      --all   add all of the lists' images if the images to add are lists

```

Reply:
Awesome, thx!